src/FsXaml.Wpf/MarkupExtensions.fs

```diff
@@ -0,0 +1,55 @@
+﻿namespace FsXaml
+open System
+open System.Windows
+open System.Windows.Data
+open System.Windows.Input
+open System.Windows.Markup
+
+module ServiceProvider =
+    let private getService<'t> (serviceProvider : IServiceProvider) =
+        serviceProvider.GetService typeof<'t> :?> 't;
+        
+    let provideValueTarget serviceProvider =
+        getService<IProvideValueTarget> serviceProvider
+
+module Reflection = 
+    let invoke source methodName arg =
+        source.GetType().GetMethod(methodName).Invoke(source, [|arg|])
+
+type internal Proxy () =
+    inherit FrameworkElement()
+    static let valueProperty = DependencyProperty.Register(
+                                        "Value", 
+                                        typeof<obj>,
+                                        typeof<Proxy>)
+
+    static member ValueProperty = valueProperty
+
+    member this.Value
+        with get() = 
+            let value = this.GetValue(Proxy.ValueProperty)
+            if obj.ReferenceEquals(value, null) then None
+            else Some(value)
+
+type HandlerExtension(observerBinding : Binding, map : obj) =
+    inherit MarkupExtension()
+    let observerBinding = observerBinding
+    let map = map
+    let proxy = Proxy()
+    let onEvent _ e =
+        match proxy.Value with
+        | Some observer -> 
+            let mapped = Reflection.invoke map "Invoke" e
+            Reflection.invoke observer "OnNext" mapped |> ignore
+        | None -> ()
+        
+    override this.ProvideValue (serviceProvider : IServiceProvider) =
+            let provideValueTarget = ServiceProvider.provideValueTarget serviceProvider
+            let dataContextBinding = Binding()
+            dataContextBinding.Source <- provideValueTarget.TargetObject
+            dataContextBinding.Mode <- BindingMode.OneWay
+            BindingOperations.SetBinding(proxy, Proxy.DataContextProperty, dataContextBinding) |> ignore
+            BindingOperations.SetBinding(proxy, Proxy.ValueProperty, observerBinding) |> ignore
+            RoutedEventHandler onEvent :>_
```

Comment:
I wrote this in c# first and it works, maybe I dumbed the conversion to f# somehow..